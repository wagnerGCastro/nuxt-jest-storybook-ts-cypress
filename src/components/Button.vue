<template>
  <div>
    <p>Times clicked: {{ count }}</p>
    <button type="button" @click="increment">{{ text }}</button>
  </div>
</template>

<script>
export default {
  name: 'ButtonDefault',
  props: {
    text: {
      type: String,
      default: 'increment',
    },
  },
  data: () => ({
    count: 0,
  }),
  methods: {
    increment() {
      this.count++;
    },
  },
};
</script>
