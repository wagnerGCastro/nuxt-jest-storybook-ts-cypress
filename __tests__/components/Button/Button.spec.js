import { render, screen, fireEvent } from '@testing-library/vue';
import { render as renderUtils } from '@vue/server-test-utils';
import Button from '@/components/Button';

it('renders correctly 1', async () => {
  const wrapper = await renderUtils(Button, { propsData: { text: 'Some Text' } });
  expect(wrapper.text()).toContain('Some Text');
  expect(wrapper).toMatchSnapshot();
});

it('renders correctly 2', () => {
  const wrapper = render(Button, { props: { text: 'Some Text' } });
  expect(wrapper.getByText('Some Text'));
});

it('increments value on click', async () => {
  // The `render` method renders the component into the document.
  // It also binds to `screen` all the available queries to interact with
  // the component.
  render(Button);

  // queryByText returns the first matching node for the provided text
  // or returns null.
  expect(screen.queryByText('Times clicked: 0')).toBeTruthy();

  // getByText returns the first matching node for the provided text
  // or throws an error.
  const button = screen.getByText('increment');

  // Click a couple of times.
  await fireEvent.click(button);
  await fireEvent.click(button);

  expect(screen.queryByText('Times clicked: 2')).toBeTruthy();
});
