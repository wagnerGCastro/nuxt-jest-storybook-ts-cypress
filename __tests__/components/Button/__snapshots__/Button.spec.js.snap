// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1 1`] = `
LoadedCheerio {
  "0": <div
    data-server-rendered="true"
  >
    <p>
      Times clicked: 0
    </p>
     
    <button
      type="button"
    >
      Some Text
    </button>
  </div>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": Array [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
